src/answers/StopwatchAnswerComponent.coffee

```diff
@@ -0,0 +1,87 @@
+React = require 'react'
+H = React.DOM
+R = React.createElement
+
+now = () -> new Date().getTime()
+
+module.exports = class StopwatchAnswerComponent extends React.Component
+  @propTypes:
+    onValueChange: React.PropTypes.func.isRequired
+
+  constructor: (props) ->
+    super
+    @state =
+      timerId: 0       # ID of the running JS timer
+      elapsedTicks: 0  # Tick count
+      editing: false   # True if in edit mode
+      editTicks: 0     # Temporary store for manual-mode tick count
+
+  # Starts a timer to update @elapsedTicks every 10 ms
+  handleStartClick: () =>
+    startTime = now() - @state.elapsedTicks # for restarts we need to fudge the startTime
+    update = () => @setState(elapsedTicks: now() - startTime)
+    @setState(timerId: setInterval(update, 10)) # create a timer and store its id in state
```

Comment:
I'm worried that using setState and relying on the accuracy of 10ms timing is going to cause problems on slow devices, especially where Javascript might lock up for 200+ms. setState isn't necc. synchronous and callback on setInterval are patchy at best. Why do we need to store elapsedTicks at all? Couldn't it be computed from now() - startTime where startTime would be the state?